src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -141,6 +141,27 @@ public function getBoolean(string $key, bool $default = false): bool
         return $this->filter($key, $default, \FILTER_VALIDATE_BOOL);
     }
 
+    /**
+     * Returns the parameter value converted to an enum.
+     *
+     * @template T of \BackedEnum
+     *
+     * @param class-string<T> $class
+     * @param ?T              $default
+     *
+     * @return ?T
+     */
+    public function getEnum(string $key, string $class, \BackedEnum $default = null): ?\BackedEnum
+    {
+        $value = $this->get($key);
+
+        if (null === $value) {
+            return $default;
+        }
+
+        return $class::tryFrom($value) ?? $default;
```

Comment:
wondering if its a good idea to return the default in case an invalid value (not a valid backend enum value) is passed? :thinking:

Reply:
I think get() behaves the same, but in what case a default should be returned then?

Reply:
Better keep it consistent with the rest of the API I think.

Reply:
With `get`, there is no invalid value.

In https://github.com/symfony/symfony/pull/48525 there is the same dilemma and the trend is to throw an exception when the value is invalid. This is particularly useful when the value comes from the query string, a BadRequest response is sent when the value is unexpected.

Reply:
What do you think to add the same parameter `$failOnInvalid`?
We could even add annotation `psalm-assert` to narrow the type

Reply:
This makes me think the exception of the `from()` should be caught to throw a `BadRequest`

Reply:
`$failOnInvalid` is added for changing the behaviour while keeping backward compatibility. I think it is better to define a behaviour for all instead of adding options.

If you return the default in case of invalid value, you cannot distinguish when the parameter is not set and when the value is invalid.

Reply:
the behavior of the method is fixed now:
- we throw an `\UnexpectedValueException` when the value is not part of the backed enum, and not a string or an int (ie: it could be an array)
- I've overridden the method in `InputBag` to throw a `BadRequestException` in that case